pkg/metrics/instance/configstore/kv/consul/mock.go

```diff
@@ -192,7 +192,12 @@ func (m *mockKV) List(prefix string, q *consul.QueryOptions) (consul.KVPairs, *c
 func (m *mockKV) Delete(key string, q *consul.WriteOptions) (*consul.WriteMeta, error) {
 	m.mtx.Lock()
 	defer m.mtx.Unlock()
+
+	// Deletes increment the ModifyIndex
 	delete(m.kvps, key)
+	m.current++
+	m.cond.Broadcast()
```

Comment:
I was very confused about why I couldn't get the delete to generate a new ModifyIndex since I could reproduce this perfectly locally with Consul. Turns out the mock consul here was accidentally missing this behavior.